pkg/prometheus/promcfg.go

```diff
@@ -853,6 +871,19 @@ func (cg *configGenerator) generateServiceMonitorConfig(
 	return cfg
 }
 
+func addressSharding(relabelings []yaml.MapSlice, shards int32) []yaml.MapSlice {
+	return append(relabelings, yaml.MapSlice{
+		{Key: "source_labels", Value: []string{"__address__"}},
+		{Key: "target_label", Value: "__hash"},
+		{Key: "modulus", Value: shards},
+		{Key: "action", Value: "hashmod"},
+	}, yaml.MapSlice{
+		{Key: "source_labels", Value: []string{"__hash"}},
```

Comment:
(nit) `__tmp_hash` instead of `__hash` as per Prometheus documentation on relabeling?